Abha_Rana/login_signup/userjwt/userjwt/Controllers/loginController.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.Tokens;
+using userjwt.CustomModels;
+using userjwt.DbModels;
+
+namespace userjwt.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class loginController : ControllerBase
+    {
+        DbModels.usersdbContext _usersdbContext;
+
+        private IConfiguration _config;
+
+        public loginController(IConfiguration config, DbModels.usersdbContext usersdbContext)
+        {
+            _config = config;
+            _usersdbContext = usersdbContext;
+        }
+
+        [AllowAnonymous]
+        [HttpPost]
+        public IActionResult CreateToken([FromBody]CustomModels.loginReq login)
+        {
+            IActionResult response = Unauthorized();
+            var user = Authenticate(login);
+
+            if (user != null)
+            {
+                var tokenString = BuildToken(user);
+                response = Ok(new { token = tokenString });
+            }
+
+            return response;
+        }
+
+        private string BuildToken(Users user)
+        {
+            var claims = new[] {
+              
+               // new Claim(JwtRegisteredClaimNames.Email, user.Email),
+                 new Claim(JwtRegisteredClaimNames.Sub, user.Username),
+                 //  new Claim("Role", user.Role),
+                  // new Claim("Address", user.Address),
```

Comment:
How are you doing role based authentication without having role in claims

Reply:
did role-based authentication. added role in claims.